projects/plugins/jetpack/modules/sharedaddy/sharing-service.php

```diff
@@ -785,6 +784,13 @@ function sharing_display( $text = '', $echo = false ) {
 		$show = false;
 	}
 
+	// Is the post password protected?
+	$post_password = $post->post_password;
+
+	if ( '' !== $post_password ) {
```

Comment:
I'd like to suggest a different approach. Sharing buttons can have different purposes:

- You can use them to widely share a post to your Facebook / Twitter followers
- You can use them to send a link to a post via email
- You can use it to send a link via WhatsApp.

In some of those scenarios, I think sharing buttons can be useful on password-protected posts. I can imagine families using them to share things in private WhatsApp groups, for example.

However, I do agree that displaying the buttons at the bottom of a post where you do not even have the password yourself is useless.

Luckily, WordPress can help us there. We could use the `post_password_required()` function. It will allow us to display the buttons to the folks who have entered the password already and who have access to the post content. It will, however, hide the buttons when the password hasn't been entered yet. 

What do you think?

```suggestion
	if ( post_password_required( $post->ID ) ) {
```

Reply:
Yes, your suggestion sounds good! I've committed the changes you suggested. Are there any other issues  with this pr?